docs/essentials/gc/unmanaged/implementingdispose.md

```diff
@@ -1 +1,442 @@
+
+Here's the general pattern for implementing the dispose pattern for a derived class that uses a safe handle: 
+
+```csharp
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+class DerivedClass : BaseClass
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+   // Instantiate a SafeHandle instance.
+   SafeHandle handle = new SafeFileHandle(IntPtr.Zero, true);
+
+   // Protected implementation of Dispose pattern.
+   protected override void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         handle.Dispose();
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+
+      disposed = true;
+      // Call base class implementation.
+      base.Dispose(disposing);
+   }
+}
+```
+
+> **Note**
+>
+> The previous example uses a [SafeFileHandle]()http://dotnet.github.io/api/Microsoft.Win32.SafeHandles.SafeFileHandle.html object to illustrate the pattern; any object derived from `SafeHandle` could be used instead. Note that the example does not properly instantiate its `SafeFileHandle` object.
+
+Here's the general pattern for implementing the dispose pattern for a derived class that overrides Object.Finalize:
+
+```csharp
+using System;
+
+class DerivedClass : BaseClass
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+
+   // Protected implementation of Dispose pattern.
+   protected override void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+      disposed = true;
+
+      // Call the base class implementation.
+      base.Dispose(disposing);
+   }
+
+   ~DerivedClass()
+   {
+      Dispose(false);
+   }
+}
+```
+
+> **Note**
+>
+> In C#, you override `Object.Finalize` by defining a `destructor`.
+
+
+## Using safe handles
+
+Writing code for an object's finalizer is a complex task that can cause problems if not done correctly. Therefore, we recommend that you construct [System.Runtime.InteropServices.SafeHandle](http://dotnet.github.io/api/System.Runtime.InteropServices.SafeHandle.html) objects instead of implementing a finalizer.
+
+Classes derived from the 'System.Runtime.InteropServices.SafeHandle' class simplify object lifetime issues by assigning and releasing handles without interruption. They contain a critical finalizer that is guaranteed to run while an application domain is unloading. The following derived classes in the Microsoft.Win32.SafeHandles namespace provide safe handles: 
```

Comment:
@svick Good question, w/rt appdomains. Two thoughts:

- What happens for processes with just a single domain, on process shutdown?
- Maybe the doc should be worded in such a way as to describe both runtimes that support appdomains and those that don't.